core/types/transaction.go

```diff
@@ -185,7 +186,11 @@ func (tx *Transaction) UnmarshalJSON(input []byte) error {
 	return nil
 }
 
-func (tx *Transaction) Data() []byte       { return common.CopyBytes(tx.data.Payload) }
+func (tx *Transaction) Data() []byte { return common.CopyBytes(tx.data.Payload) }
+func (tx *Transaction) SetData(data []byte) {
```

Comment:
IMO, transaction data is supposed to be immutable after transaction is created. Exposing `SetData`  breaks this design.